internal/ratelimit/throttle_test.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// It is performing integration tests for throttle.go
+// Set up several test cases where we have N goroutines simulating the arrival of
+// packets at a given rate, asking a limiter when to admit them.
+// limiter can accept the capacity size packets. After that, it will wait until
```

Comment:
is it based on size or number ? If it is number then should be "can accept number of packets equivalent to capacity."